File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {
	var answer int
	var recursive func(*TreeNode, int)
	recursive = func(node *TreeNode, level int) {
		if node == nil {
			if level > answer {
				answer = level
			}
			return
		}
		recursive(node.Left, level+1)
		recursive(node.Right, level+1)
	}
	recursive(root, 0)
	return answer
}

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        return recursive(root);
    }

    private int recursive(TreeNode node) {
        if (node == null) {
            return 0;
        }
        int left = recursive(node.left);
        int right = recursive(node.right);
        return left > right ? left + 1 : right + 1;
    }
}